packages/lesswrong/components/posts/TableOfContents/TableOfContents.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { PureComponent, Component } from 'react';
+import { Components, registerComponent } from 'meteor/vulcan:core';
+import { withStyles } from '@material-ui/core/styles';
+import Hidden from '@material-ui/core/Hidden';
+import withErrorBoundary from '../../common/withErrorBoundary'
+
+const styles = theme => ({
+  stickyContainer: {
+    position: "absolute",
+    right: "100%",
+    marginRight: 60,
+    top: 120,
+    width: 240,
+    height: "100%",
+  },
+  
+  stickyBlock: {
+    position: "sticky",
+    fontSize: 12,
+    top: 100,
+    lineHeight: 1.0,
+  },
+});
+
+// Context used to share a ref used to share the table of contents between the
+// ToC itself, and the Header. The Header uses the ToC to change its icon (if
+// a ToC is present) and to put the ToC inside NavigationMenu; it needs this
+// Context because it doesn't have access to the post, which is on the wrong
+// side of a whole lot of plumbing.
+//
+// The ref is to a function setToC, which puts the ToC in the state of Layout.
```

Comment:
Hmm, I am used to `ref` referring to "DOM-ref" in the React context. Maybe say "reference" or something else that makes it clear you're not talking about DOM-refs.